pages/api/calendar/add.js

```diff
@@ -0,0 +1,24 @@
+const data = require('../../../backend/data/index.js');
+
+module.exports = async (req, res) => {
+    await data.initIfNotStarted();
+    if (req.method === 'POST') {
+        const authStatus = await data.auth.checkAnyUser(req.headers['authorization'], res);
+        if (authStatus) {
+            res.setHeader('Content-Type', 'application/json');
+            
+            // Get the Access Token from the request headers
+            const token = req.headers['authorization'].split(' ')[1];
```

Comment:
there doesn't seem to be any error handling if the authorization header is undefined. calling split on it would cause a crash if the header was missing